### ISSUE TITLE
cannot read property info of undefined with version 4.2.1

### ISSUE THREAD

**Issue by vidarc** [jantimon/html-webpack-plugin#1425]
## Expected behaviour

It should build successfully

## Current behaviour

production build errors out with the following message (some paths edited to protect the innocent):
```
-  Building for production as library (commonjs,umd,umd-min)...
 ERROR  TypeError: Cannot read property 'info' of undefined
TypeError: Cannot read property 'info' of undefined
    at /node_modules/html-webpack-plugin/index.js:590:69
    at Array.filter (<anonymous>)
    at HtmlWebpackPlugin.htmlWebpackPluginAssets (/node_modules/html-webpack-plugin/index.js:580:57)
    at /node_modules/html-webpack-plugin/index.js:164:29
    at _next0 (eval at create (/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at eval (eval at create (/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:19:1)
    at /node_modules/@intervolga/optimize-cssnano-plugin/index.js:115:11
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

this is due to this commit: https://github.com/jantimon/html-webpack-plugin/commit/119252a381bf43dea37c1be64f90c10bebc21302
it builds fine with version 4.2.0 of html-webpack-plugin

logged out exactly what it was erroring out on as well. don't know if it's map files in general, or just this one:
```
chunk ddk4FFPKp.css
comp {
  name: 'ddk4FFPKp.css',
  source: { source: [Function: source], size: [Function: size] },
  info: {}
}
chunk ddk4FFPKp.common.js
comp {
  name: 'ddk4FFPKp.common.js',
  source: ConcatSource {
    children: [
      [RawSource],
      '\n//# sourceMappingURL=ddk4FFPKp.common.js.map'
    ]
  },
  info: {}
}
chunk ddk4FFPKp.css.map
comp undefined
```

## Environment

Tell us which operating system you are using, as well as which versions of Node.js, npm, webpack, and html-webpack-plugin. Run the following to get it quickly:

```
node: 12.16.2
webpack: 4.42.1
yarn: 1.22.4
html-webpack-plugin: 4.2.1
```

### Config

Copy the minimal `webpack.config.js` to produce this issue:

```js
module.exports = {
  entry: 'app.js',
  output: {
    path: 'dist',
    filename: 'index_bundle.js'
  },
  module: {
    rules: [
      ...
    ]
  }
  plugins: [
      new HtmlWebpackPlugin({
        filename: `${path.basename(parentDir)}.html`,
        publicPath: process.env.NODE_ENV === 'production' ? `${applicationPublicPath}/` : '',
        appName: `${fullyQualifiedAppName}_${process.env.BuildHash}`,
        template: 'body.hbs',
        inject: false,
      }),    ...
  ]
}
```

Copy your template file if it is part of this issue:

```html
<!-- Start app fragment -->
<div>
</div>

{{#each htmlWebpackPlugin.files.css }}
<link href="{{../htmlWebpackPlugin.options.publicPath}}{{this}}" rel="stylesheet" />
{{/each}}

{{#each htmlWebpackPlugin.files.js }}
<script type="text/javascript">
    window.router.registerScript("{{../htmlWebpackPlugin.options.appName }}", "{{../htmlWebpackPlugin.options.publicPath }}{{{this}}}");
</script>
{{/each}}
<script type="text/javascript">
    window.bootstrap.then(() => window.router.initializeApp("{{htmlWebpackPlugin.options.appName}}"));
</script>
<!-- End app fragment -->
```

## Relevant Links

- If your project is public, link to the repo so we can investigate directly.
- **BONUS POINTS:** Create a [minimal reproduction](http://stackoverflow.com/help/mcve) and upload it to GitHub. This will get you the fastest support.

**Comment by jantimon** [comment]
Hey thanks for reporting!

Which devtool are you using?

**Comment by jantimon** [comment]
Can you try if this change would fix your problem?

https://github.com/jantimon/html-webpack-plugin/pull/1426

**Comment by jantimon** [comment]
sorry I meant which webpack devtool setting https://webpack.js.org/configuration/devtool/ are you using?

**Comment by jantimon** [comment]
Okay no worries - I'll apply the patch for now - if you manage somehow to find out how we might unit test this please let me know

**Comment by vidarc** [comment]
https://github.com/vuejs/vue-cli/blob/f5135d41a53ebcd645ded3aa556260412b8cf225/packages/%40vue/cli-service/lib/config/prod.js#L6

looks like 'source-map' for production

**Comment by jantimon** [comment]
Strange I tried to create a unit test but it works also without the patch:

```
  it('works with source-maps', done => {
    testHtmlPlugin({
      mode: 'production',
      devtool: 'source-map',
      entry: path.join(__dirname, 'fixtures/index.js'),
      output: {
        path: OUTPUT_DIR
      },
      plugins: [new HtmlWebpackPlugin()]
    }, ['<script src="main.js"'], null, done);
  });

```

**Comment by jantimon** [comment]
I released the patch as `html-webpack-plugin v4.2.2`

**Comment by jantimon** [comment]
Please reopen if the bug is still happening for 4.2.2
